scripts/src/quay/linkimages.py

```diff
@@ -0,0 +1,133 @@
+
+    If the verifier version image is already linked to the specified tag no action is taken.
+
+    The default of link-tag is set to "test" to avoid accidental updates to latest.
+
+    The auth token to enable the script to link tags in quay must be set in an environment variable QUAY_AUTH_TOKEN.
+    For a github action this must be set as a repository secret.
+    For information on auth token required see:
+    https://access.redhat.com/documentation/en-us/red_hat_quay/3/html/red_hat_quay_api_guide/using_the_red_hat_quay_api
+
+results:
+    A message indicating the outcome.
+    exit code 1 if version image was not found.
+
+"""
+
+import requests
+import json
+import sys
+import argparse
+import os
+from retry import retry
+
+sys.path.append('./scripts/src/')
+from release import releasechecker
+
+# Quay API docs: https://access.redhat.com/documentation/en-us/red_hat_quay/3/html/red_hat_quay_api_guide/index
+# Quay Swagger API: https://docs.quay.io/api/swagger/
+#
+
+defaultLinkTag= "test"
+tagUrl = 'https://quay.io/api/v1/repository/redhat-certification/chart-verifier/tag/'
+
+# try every 15 seconds for 15 minutes
+@retry(Exception,tries=60, delay=15)
+def getImageId(tagValue,doRetry):
+
+    print(f"[INFO] look for tag : {tagValue}, retry : {doRetry}")
+    tagUrl = 'https://quay.io/api/v1/repository/redhat-certification/chart-verifier/tag/'
+
+    get_params = {'onlyActiveTags' : 'true','specificTag' : tagValue }
+
+    response = requests.get(tagUrl,params=get_params)
+
+    imageId = ""
+    if response.status_code > 201:
+        print(f"[Error] Error getting tags from quay : status_code={response.status_code}")
+    else:
+        tags = json.loads(response.text)
+        print("[INFO] loaded the tags")
+        for tag in tags["tags"]:
+            if tag['name'] == tagValue:
+                imageId = tag['image_id']
+                print(f"[INFO] Found tag {tagValue}. image_id : {imageId}")
+                break
+            else:
+                print(f"[INFO] ignore tag {tag['name']}")
+
+        if not imageId and doRetry:
+            print(f"[INFO] {tagValue} not found. Retry!")
+            raise Exception(f"Image {tagValue} not found")
+
+    return imageId
+
+def linkImage(linkImage,linkTag):
+
+    print(f"[INFO] Update {linkTag} to point to {linkImage}")
+    auth_token = os.environ.get('QUAY_AUTH_TOKEN')
+    if auth_token is None:
+        print("[ERROR] repository secret QUAY_AUTH_TOKEN not set")
+        return False
+
+    quay_token = f"Bearer {auth_token}"
+    put_header = {'content-type': 'application/json','Authorization': quay_token}
+
+    puturl = tagUrl + linkTag
+    putData = {'image': linkImage}
+    putOut = requests.put(puturl,data=json.dumps(putData), headers=put_header)
+    print(f"[INFO] Update link response code : {putOut.status_code}")
+    print(f"[INFO] Update link response : {putOut.text}")
+
+    return putOut.status_code == 200 or putOut.status_code == 201
```

Comment:
could use: return putOut.status_code in [200, 201]

Reply:
Much nicer, thank you.